### PR TITLE
feat(crons): Allow max_workers to be configured for consumer

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: src/sentry/consumers/__init__.py [hunk 1]
```diff
@@ -102,6 +102,12 @@ def ingest_monitors_options() -> List[click.Option]:
             default=10,
             help="Maximum time spent batching check-ins to batch before processing in parallel.",
         ),
+        click.Option(
+            ["--max-workers", "max_workers"],
+            type=int,
+            default=None,
```
Comment (evanpurkhiser) [inline src/sentry/consumers/__init__.py:108]: not 100% positive I can set `None` here

File: src/sentry/monitors/consumers/monitor_consumer.py [hunk 2]
```diff
@@ -880,18 +882,23 @@ def __init__(
...
         if mode == "parallel":
             self.parallel = True
+            self.worker = ThreadPoolExecutor(max_workers=max_workers)
 
         if max_batch_size is not None:
             self.max_batch_size = max_batch_size
         if max_batch_time is not None:
             self.max_batch_time = max_batch_time
 
     def create_paralell_worker(self, commit: Commit) -> ProcessingStrategy[KafkaPayload]:
```
Comment (wedamija) [inline src/sentry/monitors/consumers/monitor_consumer.py:896]: Nit: `parallel`

File: src/sentry/monitors/consumers/monitor_consumer.py [hunk 3]
```diff
@@ -865,6 +862,11 @@ class StoreMonitorCheckInStrategyFactory(ProcessingStrategyFactory[KafkaPayload]
     Does the consumer process unrelated check-ins in parallel?
     """
 
+    worker: Executor | None = None
+    """
+    The Executor to use whne running in parallel mode
```
Comment (wedamija) [inline src/sentry/monitors/consumers/monitor_consumer.py:867]: Nit: when

File: src/sentry/consumers/__init__.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/consumers/__init__.py:108]: I think the type only applies to input, but should be easy enough just to test quickly locally

**Review by wedamija — APPROVED** [review]

File: src/sentry/monitors/consumers/monitor_consumer.py (hunk 2 above)
Comment (evanpurkhiser) [reply src/sentry/monitors/consumers/monitor_consumer.py:896]: I'll follow up on this. I was spelling it wrong all over the place lol

File: src/sentry/consumers/__init__.py (hunk 1 above)
Comment (evanpurkhiser) [reply src/sentry/consumers/__init__.py:108]: Tested this locally. Works as expected yeah.

File: src/sentry/monitors/consumers/monitor_consumer.py (hunk 2 above)
Comment (evanpurkhiser) [reply src/sentry/monitors/consumers/monitor_consumer.py]: nvm just fixing here

**Pull request merged**